### PR TITLE
#1830 update scorecard to v5 (gh action 2.4.0)

### LINKED ISSUE #1830: scorecard: update to version 5
The openSSF scorecard execution is currently running with version 4 of scorecard

Once v5 goes GA, we should update to v5 (currently at v5.0.0-rc2)

See https://github.com/ossf/scorecard/releases

### PR DESCRIPTION
**Pull request by planetf1**
- Updates to use scorecard v5
 - re-pinned actions in scorecard.yaml to latest levels
 - fixed report of unpinned dependencies in unix.yml
 - enabled publishing of scorecard results

Note: Results are published by the openssf team at https://scorecard.dev/viewer/?uri=github.com%2Fopen-quantum-safe%2Fliboqs
Enabling our own scan is recommended by openssf, allows us to enable badges, and permits some tweaking of rules
The current scan results are [here](https://scorecard.dev/viewer/?uri=github.com%2Fopen-quantum-safe%2Fliboqs)

Fixes #1830 

n/a

* [ ] Does this PR change the input/output behaviour of a cryptographic algorithm (i.e., does it change known answer test values)?  (If so, a version bump will be required from *x.y.z* to *x.(y+1).0*.)
* [ ] Does this PR change the list of algorithms available -- either adding, removing, or renaming? Does this PR otherwise change an API? (If so, PRs in fully supported downstream projects dependent on these, i.e., [oqs-provider](https://github.com/open-quantum-safe/oqs-provider) will also need to be ready for review and merge by the time this is merged.)

No change to crypto. This is build pipeline only.

### CODE REVIEW

File: .github/workflows/scorecard.yml
Comment (baentsch) [inline .github/workflows/scorecard.yml:8]: Err -- please tell me that this is not a cronjob aimed at creating (just) the impression of "maintained"-ness, @planetf1 : If it were, I'd violently demand this to be dropped: If no-one maintains the project this fact should become visible. Otherwise this whole scorecard thing really is just a hoax.

File: .github/workflows/scorecard.yml
Comment (baentsch) [inline .github/workflows/scorecard.yml:15]: Testing only code should not be in "main" should it be? Otherwise, please document with some more words how this would be used in testing.

File: .github/workflows/scorecard.yml
Comment (baentsch) [inline .github/workflows/scorecard.yml:55]: Well, it seems this value is pointless given the results are published anyway, so what's the difference between true and false here?

**Review by baentsch — DISMISSED** [review]
Please re-base and check the single comments.

File: .github/workflows/scorecard.yml
Comment (dstebila) [reply .github/workflows/scorecard.yml:8]: If that is indeed what this action is doing, then agreed -- having an artificial commit to meet a metric seems like a bad idea.

Comment (planetf1) [reply .github/workflows/scorecard.yml:8]: This code **isn't performing a commit**, it's just [triggering if the branch protection rules are updated](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#branch_protection_rule) so that the scorecard can be refreshed. 

The comment isn't the best placed and has likely moved a little after merges - it's a generic comment (or could be removed).

However the comment is accurate, in that the [check it refers to](https://github.com/ossf/scorecard/blob/main/docs/checks.md#maintained) i is looking at recent commits and using that to determine an overall score. There is no particular threshold. .. and yes it can be gamed by introducing the noise of a unnecessary commit.

I agree the whole area of these 'heuristics' in general is debated a lot (see the [chaoss](https://chaoss.community/) community!)

Some organizations may use these metrics to understand more about a project. Overall we score fairly highly...

liboqs is in fact being scanned by openssf directly - the results are [here](https://securityscorecards.dev/viewer/?uri=github.com/open-quantum-safe/liboqs)

The action allows a few more checks to work, enables badging, raises events (security) for tracking - and in fact is needed for smaller projects they've yet to scan.

File: .github/workflows/scorecard.yml
Comment (planetf1) [reply .github/workflows/scorecard.yml:15]: I can clean up the comment.

By 'testing' I meant 'testing of the action' - the intent with scorecard is to only run it on code merged into main. However when we were getting started with it, and particularly the requirement to mitigate some of the issues,  the checks were done during a PR for expediency

File: .github/workflows/scorecard.yml
Comment (planetf1) [reply .github/workflows/scorecard.yml:55]: true means the results from this scan are published, rather than a [weekly scan done by the openssf team]. This includes a few extra checks, access to a badge, and raising of any issues found in the 'security tab'

Initially they weren't doing this on any of our repos. today AFAIK it's liboqs only.

**Comment by planetf1** [comment]
The notice above relates to the scorecard analysis being available as entries under the 'security' tab.

File: .github/workflows/scorecard.yml
Comment (SWilson4) [inline .github/workflows/scorecard.yml:28]: The hash has updated here, but the pin comment remains the same—is this still v4?

File: .github/workflows/scorecard.yml
Comment (SWilson4) [inline .github/workflows/scorecard.yml:55]: Same comment as above: is this still v4?

**Review by SWilson4 — APPROVED** [review]
Looks good to me. I don't think it's an issue to merge this before we release 0.11.0, since it only touches CI (and even that only minimally), but it would be good to get a second opinion on that.

File: .github/workflows/scorecard.yml
Comment (planetf1) [reply .github/workflows/scorecard.yml:28]: yes. There's some info as to how this works in [PROCEDURES.md](https://github.com/open-quantum-safe/liboqs/blob/main/docs/PROCEDURES.md#github-actions)

File: .github/workflows/scorecard.yml
Comment (planetf1) [reply .github/workflows/scorecard.yml:55]: as above

**Comment by planetf1** [comment]
@SWilson4 though of course I want to get this merged, I'd say leave until after release. It just feels with days to go we should minimize any unncessary changes so seems a good general practice

**Review by dstebila — APPROVED** [review]

**Pull request merged**
